File: app/src/main/java/com/example/fitnesssdemo/WalkActivity.java
package com.example.fitnesssdemo;

import android.content.Context;
import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;
import android.widget.Toast;

public class WalkActivity extends AppCompatActivity implements SensorEventListener {

    TextView steps;
    TextView km_run;
    TextView kcal_run;
    SensorManager sensorManager;
    Sensor sensor;
    boolean running;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_walk);

        steps = (TextView) findViewById ( R.id.stepsValue);
        km_run = (TextView) findViewById ( R.id.km_run);
        kcal_run = (TextView) findViewById ( R.id.kcal_run);

        sensorManager = (SensorManager) getSystemService ( Context.SENSOR_SERVICE);
    }

    @Override
    protected void onResume() {
        super.onResume ();
        running = true;
        Sensor countSensor = sensorManager.getDefaultSensor ( sensor.TYPE_ALL );
        if(countSensor != null){
            sensorManager.registerListener ( this,countSensor,SensorManager.SENSOR_DELAY_UI );
        }else {
            Toast.makeText ( this,"SENSOR NOT FOUND",Toast.LENGTH_SHORT ).show ();
        }
    }

    @Override
    protected void onPause() {
        super.onPause ();
        running = false;
    }

    @Override
    public void onSensorChanged(SensorEvent event) {

        if (running){
            steps.setText ( String.valueOf ( event.values[0] ) );
            km_run.setText ( String.valueOf ( event.values[0]*0.02 ));
            kcal_run.setText ( String.valueOf (((event.values[0]*30)/(100000)) ) );
        }
    }

    @Override
    public void onAccuracyChanged(Sensor sensor, int accuracy) {

    }
}

File: app/src/main/java/com/example/fitnesssdemo/WalkActivity.kt
package com.example.fitnesssdemo
import android.content.Context
import android.content.Intent
import android.hardware.Sensor
import android.hardware.SensorEvent
import android.hardware.SensorEventListener
import android.hardware.SensorManager
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.ImageButton
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_walk.*

class WalkActivity : AppCompatActivity(), SensorEventListener {

    internal var linear_walk: LinearLayout? = null

    var running = false
    var sensorManager:SensorManager? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        linear_walk = findViewById<View>(R.id.linear_walk) as LinearLayout

        sensorManager = getSystemService(Context.SENSOR_SERVICE) as SensorManager

        linear_walk!!.setOnClickListener(View.OnClickListener {
            val i = Intent(this@WalkActivity, MainActivity::class.java)
            startActivity(i)
        })
    }

    override fun onResume() {
        super.onResume()
        running = true
        var stepsSensor = sensorManager?.getDefaultSensor(Sensor.TYPE_STEP_COUNTER)

        if (stepsSensor == null) {
            Toast.makeText(this, "No Step Counter Sensor !", Toast.LENGTH_SHORT).show()
        } else {
            sensorManager?.registerListener(this, stepsSensor, SensorManager.SENSOR_DELAY_UI)
        }
    }

    override fun onPause() {
        super.onPause()
        running = false
        sensorManager?.unregisterListener(this)
    }

    override fun onAccuracyChanged(p0: Sensor?, p1: Int) {
    }

    override fun onSensorChanged(event: SensorEvent) {
        if (running) {
            stepsValue.setText("" + event.values[0])
            km_run.setText("" + event.values[0]*0.02)
            kcal_run.setText("" + ((event.values[0]*30)/(100000)))
        }
    }
}